src/support/libsupportinit.c

```diff
@@ -51,6 +52,31 @@ void libsupport_init(void)
     }
 }
 
+#ifdef __POSIX__
+  // Raise the open file descriptor limit.
+  {
+    struct rlimit rl;
+    if (getrlimit(RLIMIT_NOFILE, &rl) == 0 && rl.rlim_cur != rl.rlim_max) {
+      // Do a binary search for the limit.
+      rlim_t min = rl.rlim_cur;
+      rlim_t max = 1 << 20;
+      // But if there's a defined upper bound, don't search, just set it.
+      if (rl.rlim_max != RLIM_INFINITY) {
+        min = rl.rlim_max;
+        max = rl.rlim_max;
+      }
+      do {
+        rl.rlim_cur = min + (max - min) / 2;
+        if (setrlimit(RLIMIT_NOFILE, &rl)) {
+          max = rl.rlim_cur;
+        } else {
+          min = rl.rlim_cur;
+        }
+      } while (min + 1 < max);
+    }
+  }
```

Comment:
This entire code block is outside a function, so when the correct define is used it causes a build error. I am guessing you meant for this to be included inside `libsupport_init` above?